File: backend/jobporal/src/main/java/com/project/jobportal/models/NotiUserKey.java
package com.project.jobportal.models;

import jakarta.persistence.Embeddable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Embeddable
public class NotiUserKey {
    private Long IdNoti;
    private Long IdUser;
}

File: frontend_app/job_portal/lib/models/noti_user_key.dart
class NotiUserKey {
  int? idNoti;
  int? idUser;

  NotiUserKey({this.idNoti, this.idUser});

  NotiUserKey.fromJson(Map<String, dynamic> json) {
    idNoti = json['idNoti'];
    idUser = json['idUser'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idNoti'] = this.idNoti;
    data['idUser'] = this.idUser;
    return data;
  }
}
